### PR TITLE
Port the main app to Harmony 1.06.

### LINKED ISSUE #158: Update to Harmony 1.06

### PR DESCRIPTION
**Pull request by nomis52**
This also fixes a bug in reset.c

Closes #158

### CODE REVIEW

File: firmware/src/usb_descriptors.c [hunk 1]
```diff
@@ -143,7 +144,7 @@ static const uint8_t g_config_descriptor[] = {
   USB_DESCRIPTOR_ENDPOINT,  // Endpoint Descriptor
   0x2 | USB_EP_DIRECTION_IN,  // EndpointAddress ( EP2 IN INTERRUPT)
   USB_TRANSFER_TYPE_INTERRUPT,  // Attributes type of EP (INTERRUPT)
-  0x0A, 0x00,  // Max packet size of this EP
+  0x10, 0x00,  // Max packet size of this EP
```
Comment (peternewman) [inline firmware/src/usb_descriptors.c]: I take it this is intentional?

**Comment by peternewman** [comment]
LGTM, just one question.

File: firmware/src/usb_descriptors.c (hunk 1 above)
Comment (nomis52) [inline firmware/src/usb_descriptors.c]: Yup, 0x10 is 16 bytes which is one of the standard sizes. I'm not sure why the original plugin generated a max size of 10 bytes, that seems wrong :).

Comment (peternewman) [inline firmware/src/usb_descriptors.c]: Okay

**Comment by peternewman** [comment]
LGTM

**Pull request merged**